workflows/node/shared/forms.py

```diff
@@ -34,9 +34,14 @@ def node_type_selector(node_type: str) -> Choice:
         for type in netbox.get_device_types()
         if type.manufacturer.name == node_type
     }
-    return Choice("TypesEnum", zip(types.keys(), types.items()))  # type:ignore
+    return Choice("NodeTypeEnum", zip(types.keys(), types.items()))
 
 
 def node_status_selector() -> Choice:
     statuses = NodeStatus.list()
-    return Choice("NodeStatusEnum", zip(statuses, statuses))  # type:ignore
+    return Choice("NodeStatusEnum", zip(statuses, statuses))
+
+
+SiteChoice: TypeAlias = cast(type[Choice], site_selector())
+NodeRoleChoice: TypeAlias = cast(type[Choice], node_role_selector())
+NodeStatusChoice: TypeAlias = cast(type[Choice], node_status_selector())
```

Comment:
If new netbox entities are created they wont show up in these choices until user restarts the app. Not sure if that's intended?

Reply:
I noticed when I was verifying the workflows 😊 I moved the generation of all dynamic types back inside the input form generator(s).